### PR TITLE
add normalization for Vector/VectorView

### PR DESCRIPTION
**Pull request by kwitaechong**
add normalization of Vector/VectorView for usage in other application

### CODE REVIEW

File: unit_test/tstBatchedLinearAlgebra.hpp [hunk 1]
```diff
@@ -1021,6 +1021,15 @@ void vectorTest()
     EXPECT_EQ( x.stride( 0 ), 1 );
     EXPECT_EQ( x.extent( 0 ), 3 );
 
+    // normalize
+    auto x_copy = x;
```
Comment (streeve) [inline unit_test/tstBatchedLinearAlgebra.hpp:1025]: Why make the copy here?

File: src/Picasso_BatchedLinearAlgebra.hpp [hunk 2]
```diff
@@ -1596,6 +1596,16 @@ struct Vector
         return *this;
     }
 
+    // Normalization
+    KOKKOS_INLINE_FUNCTION
+    Vector& normalize()
+    {
+        auto norm2 = Kokkos::sqrt( ~( *this ) * ( *this ) );
+        for ( int i = 0; i < N; ++i )
```
[the remark this answers is not in the document]
Comment (kwitaechong) [reply src/Picasso_BatchedLinearAlgebra.hpp:1604]: done

File: unit_test/tstBatchedLinearAlgebra.hpp (hunk 1 above)
Comment (kwitaechong) [reply unit_test/tstBatchedLinearAlgebra.hpp:1025]: does not want to change the original `x` because it is used in the followings..

Comment (streeve) [reply unit_test/tstBatchedLinearAlgebra.hpp:1025]: But this is a shallow copy, isn't it?

Comment (streeve) [reply unit_test/tstBatchedLinearAlgebra.hpp:1025]: No, I'm wrong

**Review by streeve — APPROVED** [review]

**Pull request merged**